client/trino-jdbc/src/main/java/io/trino/jdbc/DriverInfo.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.jdbc;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.URL;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.io.Resources.getResource;
+import static java.lang.Integer.parseInt;
+
+final class DriverInfo
+{
+    static final String DRIVER_NAME;
+    static final String DRIVER_VERSION;
+    static final int DRIVER_VERSION_MAJOR;
+    static final int DRIVER_VERSION_MINOR;
+
+    static {
+        Properties properties = new Properties();
+        URL url = getResource(DriverInfo.class, "driver.properties");
+        try (InputStream in = url.openStream()) {
+            properties.load(in);
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+
+        DRIVER_NAME = properties.getProperty("driverName", "unknown");
+        DRIVER_VERSION = properties.getProperty("driverVersion", "unknown");
+        Matcher matcher = Pattern.compile("^(\\d+)(\\.(\\d+))?($|[.-])").matcher(DRIVER_VERSION);
+        if (!matcher.find()) {
+            DRIVER_VERSION_MAJOR = 0;
+            DRIVER_VERSION_MINOR = 0;
```

Comment:
As above, we don't need to be as flexible here anymore. `verify(matcher.find()` should probably be ok

Reply:
Fixed